src/target/cortexm.c

```diff
@@ -86,19 +86,24 @@ typedef struct cortexm_priv {
 
 /* Register number tables */
 static const uint32_t regnum_cortex_m[CORTEXM_GENERAL_REG_COUNT] = {
-	0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, /* standard r0-r15 */
-	0x10,                                                 /* xpsr */
-	0x11,                                                 /* msp */
-	0x12,                                                 /* psp */
-	0x14,                                                 /* special */
+	0U, 1U, 2U, 3U, 4U, 5U, 6U, 7U, 8U, 9U, 10U, 11U, 12U, 13U, 14U, 15U, /* r0-r15 */
+	0x10U,                                                                /* xpsr */
+	0x11U,                                                                /* msp */
+	0x12U,                                                                /* psp */
+	0x14U,                                                                /* special */
+};
+
+static const uint32_t regnum_cortex_m_trustzone[CORTEXM_TRUSTZONE_REG_COUNT] = {
+	0x18U, 0x19U, /* Non-secure msp + psp */
+	0x1aU, 0x1bU, /* Secure msp + psp */
 };
 
 static const uint32_t regnum_cortex_mf[CORTEX_FLOAT_REG_COUNT] = {
-	0x21,                                           /* fpscr */
-	0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, /* s0-s7 */
-	0x48, 0x49, 0x4a, 0x4b, 0x4c, 0x4d, 0x4e, 0x4f, /* s8-s15 */
-	0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57, /* s16-s23 */
-	0x58, 0x59, 0x5a, 0x5b, 0x5c, 0x5d, 0x5e, 0x5f, /* s24-s31 */
+	0x21U,                                                  /* fpscr */
+	0x40U, 0x41U, 0x42U, 0x43U, 0x44U, 0x45U, 0x46U, 0x47U, /* s0-s7 */
+	0x48U, 0x49U, 0x4aU, 0x4bU, 0x4cU, 0x4dU, 0x4eU, 0x4fU, /* s8-s15 */
+	0x50U, 0x51U, 0x52U, 0x53U, 0x54U, 0x55U, 0x56U, 0x57U, /* s16-s23 */
+	0x58U, 0x59U, 0x5aU, 0x5bU, 0x5cU, 0x5dU, 0x5eU, 0x5fU, /* s24-s31 */
 };
```

Comment:
Is there a good reason to keep the core register indices 32-bit unsigned? Last three times I checked, these were only used as `uint8_t`. This will reduce `.rodata` flash occupation by (20+33)*(4-1)=53*3=159 bytes. Well, plus four TZ SP regs.

Reply:
The way those registers are encoded and used here needs an overhaul and checking that nothing still tries to `sizeof()` these arrays to compute a value that goes back to GDB. If that constraint has now been fully broken, then yeah, we can shrink them to `uint8_t` arrays.

Reply:
Just done this check and yeah, that works out - so we've added one more commit to the patch series to address this. Thank you!